### ISSUE TITLE
Issue with gitversion 4.0.0-beta10

### ISSUE THREAD

**Issue by mlooise** [GitTools/GitVersion#1171]
Hi All,

I currently have a issue with the new beta-version of gitversion 4.0.0-beta10. 

We are on a gitflow branching-strategy. In version 4.0.0-beta9 i got a correct commits-since-version-source, but now in beta10 i get a very high commitcount. I had a commitcount since versionsource of 19 on my release branch in version beta9 (which was correct), but now on beta10 I get a commitcount of 82... I suspect it is counting commits on the originating branch (develop)

I see a lot of : `Merge base was due to a forward merge, next merge base is` .. which is added in version beta10 as to fix issue: https://github.com/GitTools/GitVersion/issues/1046

I added some comment, but since its closed I think it's not really noticed: https://github.com/GitTools/GitVersion/issues/1046#issuecomment-282576288

Hope someone can shed some light on this issue; I'm to much a git/gitversion novice to really pinpoint the problem or offer a solution. If possible i would gladly offer more info as of our situation.

Regards, Martin

**Comment by mlooise** [comment]
To understand how it should work. Suppose you start with a development branch when master is on 1.1. and you are on gitflow.  On development 2 commits have been done after the last release has been done.

Development would be on 1.2.0-alpha-2

Then we create a release.  (`git flow release start 1.2`)

Then we commit one more on development. This will become 1.3.0-alpha-1 ?? (this was not the case in 3.6.4 but is not the case in 4.0.0-beta).

Now we also commit on release/1.2. What should release now be ? 
1.2.0-beta-1 or 1.2.0-beta-3 ? I'm getting beta-3 in my situation; is that correct? I would assume 1.2.0-beta-1. This was also the behaviour in the old version.

**Comment by JakeGinnivan** [comment]
Thanks for reporting @mlooise, it does seem the fix is not quite right. I will have a go at fixing both issues tonight.

**Comment by JakeGinnivan** [comment]
Got to the bottom of it, will get beta 12 out shortly with a fix. In the meantime can you test https://ci.appveyor.com/project/GitTools/gitversion/build/4.0.0-PullRequest1179.1260.build.1250/artifacts to see if it fixes your issues?

**Comment by mlooise** [comment]
Hi Jake, 

Thanks for investigating my problem so thoroughly! I really appreciate the way issues are picked up by this community. I am however sad to report that i can't reproduce the problem anymore. We had this issue on our production repo and i've finished the release in the meantime so the branch on which the problem existed is now gone. I have tried to reproduce it but with no luck yet. 

In the meantime I've spend some time looking at the code and unittests and i'm quite impressed with the unittests. If i run into troubles again I will surely write a failing test to mimic it in order to fix it; very nice. 

A small question regarding the added tests in your own repo; These check the merge-base, but from beta10 onwards it looks like the endresult (fullsemver) is correct (I replayed the steps on a real repo). The tests fail in e.g. beta10, so I guess you have correctly investigated the case and the 'find-base' looks ok now in your version. 

I guess I am for now very happy! Again, if I find an issue in the future I will provide my own unittests!

Regards, Martin
